File: examples/javascript/mega_flow/pipeline.js
// Async pipeline — exercises destructuring / rest, array methods
// (map / filter / reduce / flatMap), closures, generators, switch,
// try / catch / finally, template literals, optional chaining, and
// async iterator (for-await-of).
const { persist: persistEnvelope } = require("./storage");

const dynamicStorage = () => import("./storage");

// Generator — yields every token of the tainted command in order.
function* tokenize(cmd) {
  for (const part of cmd.split(" ")) {
    if (part) yield part;
  }
}

// Async generator — re-emits tokens for for-await-of consumption.
async function* stream_tokens(cmd) {
  for (const tok of tokenize(cmd)) {
    yield tok;
  }
}

// Closure factory — returns a reducer that concatenates via template
// literal while preserving taint.
const make_joiner = (sep) => (acc, tok) => acc ? `${acc}${sep}${tok}` : tok;

async function orchestrate(envelope) {
  const { cmd, user, ...rest } = envelope; // destructuring + rest
  const extraCount = envelope.extras?.length ?? 0;
  if (extraCount < 0) throw new Error("unreachable");

  // Collect tainted tokens via async iteration.
  const tokens = [];
  for await (const tok of stream_tokens(cmd)) {
    tokens.push(tok);
  }

  // Array pipeline: map → filter → flatMap → reduce.
  const joined = tokens
    .map((s) => s.trim())
    .filter(Boolean)
    .flatMap((s) => [s])
    .reduce(make_joiner(" "), "");

  // switch/case — every branch re-binds the tainted string.
  let routed;
  switch (envelope.kind) {
    case "run":
      routed = `${joined}`;
      break;
    case "eval":
      routed = joined.trim();
      break;
    default:
      routed = joined;
  }

  // try/catch/finally — taint survives the try block.
  let valid;
  try {
    if (!routed) throw new Error("empty");
    valid = { ...rest, user, cmd: routed };
  } catch (err) {
    valid = { ...rest, user, cmd: routed ?? "" };
  } finally {
    // no-op — present so the adapter sees the finally clause.
  }

  return await persistEnvelope(valid);
}

module.exports = { orchestrate, tokenize, stream_tokens };

File: examples/java/language_gauntlet/src/main/java/gauntlet/pipeline/Pipeline.java
package gauntlet.pipeline;

import static java.util.Objects.requireNonNull;

import gauntlet.domain.Envelope;
import gauntlet.storage.Storage;
import java.util.Arrays;
import java.util.List;
import java.util.function.BinaryOperator;

public final class Pipeline {
    private Pipeline() {}

    interface CommandStage {
        String apply(String value);
    }

    static final class AuditScope implements AutoCloseable {
        @Override
        public void close() {}
    }

    static BinaryOperator<String> makeJoiner(String separator) {
        return (acc, token) -> acc.isEmpty() ? token : acc + separator + token;
    }

    static String route(Envelope envelope, String joined) {
        return switch (envelope.kind()) {
            case RUN -> "" + joined;
            case EVAL -> joined.strip();
        };
    }

    public static String orchestrate(Envelope envelope) {
        String cmd = requireNonNull(envelope.cmd());
        for (String extra : List.of(cmd)) {
            if (extra.isEmpty()) break;
        }
        while (cmd.startsWith(" ")) {
            cmd = cmd.substring(1);
        }

        String joined = Arrays.stream(cmd.split(" "))
                .map(String::trim)
                .filter(value -> !value.isEmpty())
                .reduce("", makeJoiner(" "));
        CommandStage router = value -> route(envelope, value);
        String routed = router.apply(joined);

        Envelope valid;
        try (var scope = new AuditScope()) {
            if (routed.isEmpty()) throw new IllegalArgumentException("empty");
            valid = new Envelope(envelope.kind(), routed, envelope.user(), routed.length(), envelope.extras());
        } catch (RuntimeException error) {
            valid = new Envelope(envelope.kind(), routed, envelope.user(), routed.length(), envelope.extras());
        } finally {
            // Explicit finally keeps exceptional compiler flow in the fixture.
        }
        return Storage.persist(valid);
    }
}
